# Analisa Variasi Dataset

In [3]:
from tensorflow.keras.preprocessing.image import ImageDataGenerator

TrainingImagePath='D:/face regocnation project/Face Images/Final Training Images'



train_datagen = ImageDataGenerator(
        shear_range=0.1,
        zoom_range=0.1,
        horizontal_flip=True)


test_datagen = ImageDataGenerator()


training_set = train_datagen.flow_from_directory(
        TrainingImagePath,
        target_size=(64, 64),
        batch_size=32,
        class_mode='categorical')



test_set = test_datagen.flow_from_directory(
        TrainingImagePath,
        target_size=(64, 64),
        batch_size=32,
        class_mode='categorical')

# Printing class labels for each face
test_set.class_indices

Found 244 images belonging to 16 classes.


Found 244 images belonging to 16 classes.


{'face1': 0,
 'face10': 1,
 'face11': 2,
 'face12': 3,
 'face13': 4,
 'face14': 5,
 'face15': 6,
 'face16': 7,
 'face2': 8,
 'face3': 9,
 'face4': 10,
 'face5': 11,
 'face6': 12,
 'face7': 13,
 'face8': 14,
 'face9': 15}

Pada hasil pembacaan dataset diatas terdapat 244 gambar dimana terdapat 16 variasi karakter

# Analisa Kinerja Model

In [4]:

TrainClasses=training_set.class_indices


ResultMap={}
for faceValue,faceName in zip(TrainClasses.values(),TrainClasses.keys()):
    ResultMap[faceValue]=faceName


import pickle
with open("ResultsMap.pkl", 'wb') as fileWriteStream:
    pickle.dump(ResultMap, fileWriteStream)


print("Mapping of Face and its ID",ResultMap)


OutputNeurons=len(ResultMap)
print('\n The Number of output neurons: ', OutputNeurons)

Mapping of Face and its ID {0: 'face1', 1: 'face10', 2: 'face11', 3: 'face12', 4: 'face13', 5: 'face14', 6: 'face15', 7: 'face16', 8: 'face2', 9: 'face3', 10: 'face4', 11: 'face5', 12: 'face6', 13: 'face7', 14: 'face8', 15: 'face9'}

 The Number of output neurons:  16


In [5]:

from keras.models import Sequential
from keras.layers import Convolution2D
from keras.layers import MaxPool2D
from keras.layers import Flatten
from keras.layers import Dense


classifier= Sequential()


classifier.add(Convolution2D(32, kernel_size=(5, 5), strides=(1, 1), input_shape=(64,64,3), activation='relu'))


classifier.add(MaxPool2D(pool_size=(2,2)))


classifier.add(Convolution2D(64, kernel_size=(5, 5), strides=(1, 1), activation='relu'))

classifier.add(MaxPool2D(pool_size=(2,2)))


classifier.add(Flatten())


classifier.add(Dense(64, activation='relu'))

classifier.add(Dense(OutputNeurons, activation='softmax'))



classifier.compile(loss='categorical_crossentropy', optimizer = 'adam', metrics=["accuracy"])


import time

StartTime=time.time()


classifier.fit(
                    training_set,
                    steps_per_epoch=30,
                    epochs=10,
                    validation_data=test_set,
                    validation_steps=10)

EndTime=time.time()
print("###### Total Time Taken: ", round((EndTime-StartTime)/60), 'Minutes ######')

Epoch 1/10


c:\Users\Muhammad Iqbal\AppData\Local\Programs\Python\Python311\Lib\site-packages\keras\src\trainers\data_adapters\py_dataset_adapter.py:120: UserWarning: Your `PyDataset` class should call `super().__init__(**kwargs)` in its constructor. `**kwargs` can include `workers`, `use_multiprocessing`, `max_queue_size`. Do not pass these arguments to `fit()`, as they will be ignored.
  self._warn_if_super_not_called()


 7/30 ━━━━━━━━━━━━━━━━━━━━ 1s 50ms/step - accuracy: 0.0825 - loss: 121.2472

c:\Users\Muhammad Iqbal\AppData\Local\Programs\Python\Python311\Lib\contextlib.py:155: UserWarning: Your input ran out of data; interrupting training. Make sure that your dataset or generator can generate at least `steps_per_epoch * epochs` batches. You may need to use the `.repeat()` function when building your dataset.
  self.gen.throw(typ, value, traceback)


30/30 ━━━━━━━━━━━━━━━━━━━━ 4s 39ms/step - accuracy: 0.0947 - loss: 97.0288 - val_accuracy: 0.0820 - val_loss: 5.8414
Epoch 2/10
30/30 ━━━━━━━━━━━━━━━━━━━━ 1s 26ms/step - accuracy: 0.1713 - loss: 3.4867 - val_accuracy: 0.2746 - val_loss: 2.0700
Epoch 3/10
30/30 ━━━━━━━━━━━━━━━━━━━━ 1s 25ms/step - accuracy: 0.3405 - loss: 1.9658 - val_accuracy: 0.6148 - val_loss: 1.1550
Epoch 4/10
30/30 ━━━━━━━━━━━━━━━━━━━━ 1s 27ms/step - accuracy: 0.6841 - loss: 1.1097 - val_accuracy: 0.8852 - val_loss: 0.3996
Epoch 5/10
30/30 ━━━━━━━━━━━━━━━━━━━━ 1s 25ms/step - accuracy: 0.8578 - loss: 0.4432 - val_accuracy: 0.9713 - val_loss: 0.0978
Epoch 6/10
30/30 ━━━━━━━━━━━━━━━━━━━━ 1s 29ms/step - accuracy: 0.9763 - loss: 0.1102 - val_accuracy: 1.0000 - val_loss: 0.0105
Epoch 7/10
30/30 ━━━━━━━━━━━━━━━━━━━━ 1s 28ms/step - accuracy: 0.9872 - loss: 0.0423 - val_accuracy: 0.9918 - val_loss: 0.0363
Epoch 8/10
30/30 ━━━━━━━━━━━━━━━━━━━━ 1s 29ms/step - accuracy: 0.9783 - loss: 0.0582 - val_accuracy: 0.9959 - val_loss: 0

Dari hasil evalusai training data diats bahwa model telah menunjukkan nilai akurasi yang tinggi, yaitu diatas 90%. Sehingga kinerja model dirasa dapat diandalkan.